File: js/programmers/level1/최소직사각형.js
function solution(sizes) {
  let s1 = 0;
  let s2 = 0;
  for (let size of sizes) {
    if (size[0] >= size[1]) {
      s1 = Math.max(s1, size[0]);
      s2 = Math.max(s2, size[1]);
    } else {
      s1 = Math.max(s1, size[1]);
      s2 = Math.max(s2, size[0]);
    }
  }

  return s1 * s2;
}

console.log(
  solution([
    [60, 50],
    [30, 70],
    [60, 30],
    [80, 40],
  ]),
);
console.log(
  solution([
    [10, 7],
    [12, 3],
    [8, 15],
    [14, 7],
    [5, 15],
  ]),
);
console.log(
  solution([
    [14, 4],
    [19, 6],
    [6, 16],
    [18, 7],
    [7, 11],
  ]),
);

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int s1=0;
        int s2=0;
        for (int[] size : sizes) {
            if (size[0] > size[1]) {
                s1 = Math.max(size[0], s1);
                s2 = Math.max(size[1], s2);
            } else {
                s1 = Math.max(size[1], s1);
                s2 = Math.max(size[0], s2);
            }
        }
        
        return s1 * s2;
    }
}